File: test/page-objects/streams/UI/eventTemplateSelection.ts
import Page from "../../page"
import reporter from "../../../utils/reporter"

class EventTemplateSelection extends Page {

    public static selectedTemplateName: string

    constructor() {
        super()
    }

    async selectEventTemplateAtRandom() {
        try {
            let random = Math.floor(Math.random() * 3) + 1;
            EventTemplateSelection.selectedTemplateName = await ($(`(//button[@class='primary']//preceding::h3)[${random}]`)).getText()
            reporter.addStep("info", `The template name is:  ${EventTemplateSelection.selectedTemplateName}`);
            await browser.pause(2000)
            await this.click(await $(`(//button[@class='primary'])[${random}]`))
            reporter.addStep("info", `Clicked on the Event random template.`)
        } catch (err) {
            err.message = `Error selecting the event template at random: , ${err.message} `
            throw err
        }
    }
}
export default new EventTemplateSelection

File: src/main/java/com/brandlive/qa/automation/events3/pages/admin/EventTemplateSelection.java
package com.brandlive.qa.automation.events3.pages.admin;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

import com.brandlive.qa.automation.utils.Base;

public class EventTemplateSelection extends Base {

    public static String selectedTemplateName = new String();

    public void selectEventTemplateAtRandom() {
        int random = (int) (Math.random() * 3) + 1;
        WebElement templateNameH3 = webDriver.findElement(By.xpath("(//button[@class='primary'])[" + random + "]//parent::div//parent::div/h3"));
        selectedTemplateName = templateNameH3.getText();
        WebElement templateButton = webDriver.findElement(By.xpath("(//button[@class='primary'])[" + random + "]"));
        templateButton.click();
    }

    public void returnToEvents() {
        WebElement returnToEventsButton = webDriver.findElement(By.xpath("//a[@href='/admin/events']"));
        returnToEventsButton.click();
    }

    public void selectTheTemplateForEvent(String templateName) {
        String template = "(//h3[text()='" + templateName + "']//following::div[@class='button-row']//button)[1]";
        WebElement templateNameH3 = waitForDisplayedAndEnabled(By.xpath(template), "Template Name Section.");
        templateNameH3.click();
        log.info("Selected the template: " + templateName);
    }
}
